src/cbmc/bmc.cpp

```diff
@@ -321,9 +327,13 @@ safety_checkert::resultt bmct::run(
   {
     error() << "Invalid memory model " << mm
             << " -- use one of sc, tso, pso" << eom;
-    return safety_checkert::resultt::ERROR;
+    throw 0;
```

Comment:
Suggest throwing something more structured, so a caller could catch it reliably

Reply:
What do you suggest for invalid memory model?

Reply:
I guess it should be an instance of `invalid_command_line_parametert` or something, which doesn't currently exist. I guess that's a separate PR.

Reply:
I'm a bit worried that this changes from reporting-a-user-error to throwing an exception. A wrong command-line parameter should not cause an abort.